dataset/src/androidMain/kotlin/org/jetbrains/kotlinx/dl/dataset/preprocessing/Preprocessing.kt

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2020-2022 JetBrains s.r.o. and Kotlin Deep Learning project contributors. All Rights Reserved.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the LICENSE.txt file.
+ */
+
+package org.jetbrains.kotlinx.dl.dataset.preprocessing
+
+import android.graphics.Bitmap
+import org.jetbrains.kotlinx.dl.dataset.preprocessing.bitmap.Resize
+import org.jetbrains.kotlinx.dl.dataset.preprocessing.bitmap.Rotate
+
+/**
+ * The data preprocessing pipeline presented as Kotlin DSL on receivers.
+ *
+ * Could be used to handle directory of images or one image file.
+ */
+public class Preprocessing : PreprocessingBase<Bitmap, FloatArray>() {
+}
+
+public fun preprocess(init: Preprocessing.() -> Unit): Preprocessing =
+    Preprocessing().apply(init)
+
+/** Applies [Resize] operation to resize the image to a specific size. */
+public fun Preprocessing.resize(block: Resize.() -> Unit) {
+    addOperation(Resize().apply(block))
+}
+
+public fun Preprocessing.rotate(block: Rotate.() -> Unit) {
+    addOperation(Rotate().apply(block))
+}
+
+public fun Preprocessing.toFlatArray(block: ConvertToFlatArray.() -> Unit) {
```

Comment:
The problem here is that we can't check in compile time that operations have correct input and output types which could lead to unexpected runtime errors. It seems that right now we don't need anything more than "process image" -> "convert to array" -> "process array" (or some other data type instead of an image) so maybe we should postpone this change when we have more use-cases.